### DISCUSSION TITLE
React\Http\Browser not work in async way

### DISCUSSION THREAD

**Discussion by emrancu** [reactphp/reactphp#544] [Q&A]
Code is working on macOS  with herd PHP8.2 but after deploy its not working in ubuntu server with PHP8.2.

I used the React\Http\Browser inside a Laravel Job/queue. 

Here is my code:

```php
$browser->post(
            'https://domain.com/api/validate',
            [
                'Content-Type' => 'application/json'
            ],
            json_encode([
                'data' =>  []
            ])
        )->then(function (ResponseInterface $response) {
          
            DB::beginTransaction();
            try {
               // ...... code

                DB::commit();
            } catch (Exception $exception) {
                DB::rollBack();
                Log::error(  '......');
            }
        },
            function (Exception $exception) {
                Log::error(  '-----');
            }
        );

```

Please help to findout the reason. 

Thank you

**Comment by SimonFrings** [comment]
@emrancu Thanks for bringing this up :+1:

One thing I've noticed is that you're using a blocking database client within your `then`, which is currently blocking your code. If you want to use the full potential of async and non-blocking here, you might want to use an async database client like https://github.com/friends-of-reactphp/mysql.

To be completely honest, I'm not entirely sure about the specific issue you're facing here, as you've described it simply as "not working." Have you attempted to break down the problem or can you provide more details on what is functioning correctly and what is not? This additional information would greatly help me understand the root cause of the problem.
